zebra-network/src/config.rs

```diff
@@ -188,12 +197,25 @@ impl Config {
     ///
     /// If DNS continues to fail, returns an empty list of addresses.
     async fn resolve_host(host: &str, max_retries: usize) -> HashSet<SocketAddr> {
-        for retry_count in 1..=max_retries {
-            match Config::resolve_host_once(host).await {
-                Ok(addresses) => return addresses,
-                Err(_) => tracing::info!(?host, ?retry_count, "Retrying peer DNS resolution"),
-            };
-            tokio::time::sleep(DNS_LOOKUP_TIMEOUT).await;
+        for retries in 0..=max_retries {
+            if let Ok(addresses) = Config::resolve_host_once(host).await {
+                return addresses;
+            }
+
+            if retries < max_retries {
+                tracing::info!(
+                    ?host,
+                    previous_attempts = ?(retries + 1),
+                    "Waiting {DNS_LOOKUP_TIMEOUT:?} to retry seed peer DNS resolution",
+                );
+                tokio::time::sleep(DNS_LOOKUP_TIMEOUT).await;
+            } else {
+                tracing::info!(
+                    ?host,
+                    attempts = ?(retries + 1),
+                    "Seed peer DNS resolution failed, checking for addresses from other seed peers",
+                );
+            }
         }
```

Comment:
Nitpick/Optional: refactor this to remove the if statement

```suggestion
        for retries in 1..=max_retries {
            tracing::info!(
                ?host,
                previous_attempts = ?(retries + 1),
                "Waiting {DNS_LOOKUP_TIMEOUT:?} to retry seed peer DNS resolution",
            );
            tokio::time::sleep(DNS_LOOKUP_TIMEOUT).await;

            if let Ok(addresses) = Config::resolve_host_once(host).await {
                return addresses;
            };
        }

        tracing::info!(
            ?host,
            attempts = ?(max_retries + 1),
            "Seed peer DNS resolution failed, checking for addresses from other seed peers",
        );
```

Reply:
I usually prefer to duplicate logging rather than duplicating the action outside the loop, because then it's possible for the duplicate actions to get out of sync.

(If the logging gets out of sync it doesn't matter as much.)